packages/pwa/app/routes.jsx

```diff
@@ -81,26 +81,31 @@ const routes = [
         exact: true
     },
     {
-        path: '/:locale/cart',
+        path: '/:siteAlias/:locale/cart',
```

Comment:
(this comment applies to all path changes made to this file)

Have you thought about adding "prepending locale to the urls" feature to the SDK, for example to the [Route component](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/develop/packages/pwa-kit-react-sdk/src/ssr/universal/components/route-component/index.js)

This is a very rough idea, but would it make sense to do:
```
<Route multisite={true} i18n={true} path="/cart"/>
```

The idea is to add feature flags to the route component, and let the SDK handle generating the URL structures. Instead of relying developers to remember the fact PWA path starts with site, then locale, then the rest of the path.

Reply:
the reason i think ^ might be a good idea is, what if a customer doesn't have multiple sites and doesn't like the the path pattern (cuz the pattern brings no benefit at all to those customers)? They'll need to do lots of find/replace to all the hard coded paths in the PWA.

Reply:
Just wanting to add a +1 to this idea since this would simplify how our paths look like. Also because I agree with Kev that there will be projects without multisite or multiple locales and I can see these projects not wanting to add, for example, /us/en-us/ to all of their paths.

Reply:
Alternatively, we can use _optional parameters_ for our routes. Optional parameter is not explicitly mentioned in the React Router doc. But it [mentions](https://reactrouter.com/web/api/Route/path-string-string) that the path can be anything that `path-to-regexp` library understands.

In path-to-regexp doc: https://github.com/pillarjs/path-to-regexp/tree/v1.7.0#optional

So for example, the PDP's route can be `path: '/:siteAlias?/:locale?/product/:productId'` and it will match these URLs:
- http://localhost:3000/global/en-GB/product/25599647M?pid=013742000269M
- http://localhost:3000/en-GB/product/25599647M?pid=013742000269M
- http://localhost:3000/product/25599647M?pid=013742000269M

Reply:
@vmarta The complication lies in url builders. We currently have quite a few of them such as `homeUrlBuilder`, `categoriesUrlBuilder`, `getUrlLocale`, and so forth. 

If the siteAlias become optional, every single of these functions will need to be aware of that. however, if we know the fact that siteAlias in always at the first part of the pathname, things are more simpler. 

Another example is the [getLocaleUrl](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/167/files#diff-6416d684e7371efec6ccf45c6a5a93f16b55c47245f872daf81a8da904cb364bL49), previously, we assume the shortCode from the url is the first element derived from the pathname, since we have siteAlias param added in, the shortCode will be the second element from the pathname. Imaging the siteAlias it optional, we have to come up with solution to find the locale shortCode, sure we can use regexp to match the pattern, but it is not 100% foolproof. If a customer happens to choose an alias that matches the regex, it is likely to mess up the localization selection functionality.

Reply:
If our future desire is to provide more flexibility in how projects can define routes, but still make it sane to make such a change, then I think we want to think of the right design pattern for providing that for both routes and URL builders

If we had say a strategy design (https://refactoring.guru/design-patterns/strategy) that would let us provide 2 default strategies to choose from (withSiteAndLocale, withLocale) and also an interface to let teams plug in their own strategies, that would be a nice way to show what we support by default but also provide a place where you can define your own but still keep in contained